### ISSUE TITLE
Kustomize not clearing /tmp folder

### ISSUE THREAD

**Issue by davidsbond** [kubernetes-sigs/kustomize#2201]
Hello, my team is using a combination of kustomize and kube-applier in order to achieve a gitops approach to our deployments in k8s.

We were experiencing an issue where our kube-applier instance began using an excessive amount of disk space, causing pods to become evicted.

After some investigation, I found the source of the disk usage was by kustomize storing copies of repositories it had to clone within the `/tmp` directory. For example, we had a `kustomization.yaml` file which cloned bases at different commits from a very large repository.

```
/tmp # ls
kustomize-018564761  kustomize-110336189  kustomize-317372402  kustomize-730356484
kustomize-061106146  kustomize-131723564  kustomize-544392991  kustomize-829899843
kustomize-076705691  kustomize-242860388  kustomize-559784796  kustomize-892497778
```

I'm not 100% certain, but it seems that in some scenarios the `kustomize build` command does not clean up the `/tmp` folders it creates

**Comment by Liujingfang1** [comment]
Which version of kustomize do you use? Can you verify this with the latest kustomize version?

If anyone wants to help with fixing it, the problem is from to the [Loader interface](https://github.com/kubernetes-sigs/kustomize/blob/master/api/ifc/ifc.go#L38), which has a function of `Cleanup` which aims to remove the temp directories used for remote target.

**Comment by davidsbond** [comment]
We're using `kubectl` version 1.14.2, which I think is bundled with 2.0.3

**Comment by vanou** [comment]
I want to handle this issue.

**Comment by davidsbond** [comment]
@vanou happy to try and give you more on what's happening my end if that helps

**Comment by vanou** [comment]
@davidsbond 
Sorry for slow progress. Yes, I would like to understand situation.

1. Where did you use kustomize & how?

   I saw GitHub README of kube-applier and my understanding is that

     * kube-applier regularly checks git repo which includes k8s resources in yaml/JSON format
     * when k8s resources in that repo is updated, kube-applier run `kubectl apply -f` against updated ones

   So I would like to know who runs kustomize/`kubectl apply -k`. Do you modify kube-applier for your team's demand?

**Comment by davidsbond** [comment]
Hey @vanou, we're using a modified version:

https://github.com/utilitywarehouse/kube-applier.

**Comment by vanou** [comment]
Thanks @davidsbond .

I would like to know when you run `kustomize` locally, not on kube-applier`, same issue happens or not.

**Comment by davidsbond** [comment]
@vanou I've never seen this locally myself, inspecting my `/tmp` folder I can't see anything left over. Could it be that if the kubectl process is killed while doing an `apply` the files are left over? Does `kustomize` handle interrupt signals and clean up?

**Comment by vanou** [comment]
Yes. It looks like that sending Kill signal during `kustomize` execution prevents cleanup code. When I interrupt `kustomize` during internal `git` operation, `kustomize-*` dirs are left. However, when `kustomize` succeeds or fails not interrupted , `kustomize-*` dirs get cleaned up.
I saw [source code](https://github.com/kubernetes-sigs/kustomize/blob/master/api/loader/fileloader.go#L198) and it looks good.

`kustomize` and another go program cannot handle `SIGKILL` & `SIGSTOP` because they are interruption forced by OS.

**Comment by davidsbond** [comment]
Ah, perhaps that is what is happening here then. In the error logs of `kube-applier`, I could see `signal: killed`. So it may just be a scenario we can't catch. More likely the issue is in our version of `kube-applier` that's causing `SIGKILL` to be sent to the commands.

**Comment by vanou** [comment]
OK.

If you feel this problem is solved or there is no work related to this issue `kustomize` developers should work on, please close this issu with `/close`.

**Comment by davidsbond** [comment]
Thanks, I think I will close for now :)

**Comment by venkattech3889** [comment]
I am also having the same issue, where we do not use kube-applier. folders are filing up like anything it's crossing1 lakh. how can we handle this?

**Comment by Shell32-Natsu** [comment]
@venkattech3889 What's the version of kustomize are you using?

**Comment by venkattech3889** [comment]
We are using 3.7.0

**Comment by Shell32-Natsu** [comment]
@venkattech3889 Are you using remote resources? If so, please try with latest version. The fix is #2943
